src/core/algorithms/ind/faida/faida.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "algorithms/ind/faida/inclusion_testing/iinclusion_tester.h"
+#include "algorithms/ind/faida/preprocessing/preprocessor.h"
+#include "algorithms/ind/faida/util/simple_ind.h"
+#include "algorithms/ind/ind_algorithm.h"
+#include "config/thread_number/type.h"
+
+namespace algos {
+
+class Faida : public algos::INDAlgorithm {
+private:
+    using SimpleCC = faida::SimpleCC;
+    using SimpleIND = faida::SimpleIND;
+
+    int sample_size_;
+    double hll_accuracy_;
+    bool detect_nary_;
+    config::ThreadNumType number_of_threads_;
+
+    size_t prepr_time_ = 0;
+    size_t insert_time_ = 0;
+    size_t check_time_ = 0;
+
+    std::unique_ptr<faida::IInclusionTester> inclusion_tester_;
+    std::unique_ptr<faida::Preprocessor> data_;
+
+    INDList result_;
+    DatasetsOrder dataset_order_;
+
+    // Insert rows into InclusionTester
+    void InsertRows(std::map<int, std::unordered_set<int>> const& active_tables,
+                    faida::Preprocessor const& data);
+    std::vector<SimpleIND> TestCandidates(std::vector<SimpleIND> const& candidates);
+    void RegisterInds(std::vector<SimpleIND> const& inds);
+
+public:
+    Faida();
+
+    void LoadDataInternal() final;
+    void MakeExecuteOptsAvailable() final;
+    unsigned long long ExecuteInternal() final;
+    void ResetState() final;
```

Comment:
`LoadDataInternal`/`ExecuteInternal`/`ResetState` - private
`MakeOptionsAvailable` - protected

Reply:
Interestingly, in the base class `MakeOptionsAvailable` is protected, but it is only used in the base class (`Algorithm`). I think, that it would be more correct to use 'private'.